bin/plugin/commands/performance.js

```diff
@@ -178,42 +178,27 @@ function curateResults( testSuite, results ) {
  * @param {string} testSuite                Name of the tests set.
  * @param {string} performanceTestDirectory Path to the performance tests' clone.
  * @param {string} runKey                   Unique identifier for the test run, e.g. `branch-name_post-editor_run-3`.
+ * @param {string} artifactsPath            Path to save the test artifacts at.
  *
  * @return {Promise<WPPerformanceResults>} Performance results for the branch.
  */
-async function runTestSuite( testSuite, performanceTestDirectory, runKey ) {
-	try {
-		await runShellScript(
-			`npm run test:performance -- packages/e2e-tests/specs/performance/${ testSuite }.test.js`,
-			performanceTestDirectory
-		);
-	} catch ( error ) {
-		fs.mkdirSync( './__test-results/artifacts', { recursive: true } );
-		const artifactsFolder = path.join(
-			performanceTestDirectory,
-			'artifacts/'
-		);
-		await runShellScript(
-			'cp -Rv ' + artifactsFolder + ' ' + './__test-results/artifacts/'
-		);
+async function runTestSuite(
+	testSuite,
+	performanceTestDirectory,
+	runKey,
+	artifactsPath
+) {
+	const resultsFilename = `${ runKey }.results.json`;
 
-		throw error;
-	}
-
-	const resultsFile = path.join(
-		performanceTestDirectory,
-		`packages/e2e-tests/specs/performance/${ testSuite }.test.results.json`
-	);
-	fs.mkdirSync( './__test-results', { recursive: true } );
-	fs.copyFileSync( resultsFile, `./__test-results/${ runKey }.results.json` );
-	const rawResults = await readJSONFile(
-		path.join(
-			performanceTestDirectory,
-			`packages/e2e-tests/specs/performance/${ testSuite }.test.results.json`
-		)
+	await runShellScript(
+		`npm run test:performance -- ${ testSuite } --wordpress-artifacts-path=${ artifactsPath } --results-filename=${ resultsFilename }`,
```

Comment:
if we're relying on the `ENV` is it necessary to pass this here or could the performance test read the same `ENV`?

Reply:
Yeah, I thought it would, but it looks like only the top-level process ([`node cli.js perf`](https://github.com/WordPress/gutenberg/blob/f2e31bdd0f925614ac73be9415ba5117e53617ad/.github/workflows/performance.yml#L56)) receives the `ENV` from CI. The `await runShellScript( "npm run ...` spawns a separate process that doesn't receive the same `ENV` so I needed to figure out a way to pass it down. Maybe there's a better way to share the env, but at my current level of understanding, this made the most sense.

Reply:
we can manually pass through any `ENV` we want in `runShellScript`

https://github.com/WordPress/gutenberg/blob/6434e0bc181bb1c0f5fcf48520be20f1a37afb6c/bin/plugin/lib/utils.js#L17-L26

@youknowriad would it make sense to pass `env: { ...process.env, …, ...env }` as a default? instead of `env: { …, ...env }`? would that be dangerous?

Reply:
> we can manually pass through any ENV we want in runShellScript

Yeah I've just noticed that! 😄 Made already some changes in that regard in https://github.com/WordPress/gutenberg/commit/a53065b9b9dd37366df0eb4c5dfd3d0ee55ade22. Not passing the whole parent env though - just the relevant bits. I'm curious though if just passing the whole thing down would be safe.